public/app/features/logs/components/panel/fields/FieldSelector.tsx

```diff
@@ -0,0 +1,413 @@
+    flexDirection: 'column',
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    paddingTop: 2,
+  }),
+  collapsedButton: css({
+    margin: 0,
+  }),
+};
+
+/**
+ * FieldSelector wrapper for the LogsTable visualization.
+ */
+interface LogsTableFieldSelectorProps {
+  columnsWithMeta: FieldNameMetaStore;
+  clear(): void;
+  dataFrames: DataFrame[];
+  logs: LogListModel[];
+  reorder(columns: string[]): void;
+  setSidebarWidth(width: number): void;
+  sidebarWidth: number;
+  toggle(key: string): void;
+}
+
+export const LogsTableFieldSelector = ({
+  columnsWithMeta,
+  clear,
+  dataFrames,
+  logs,
+  reorder,
+  setSidebarWidth,
+  sidebarWidth,
+  toggle,
+}: LogsTableFieldSelectorProps) => {
+  const setSidebarWidthWrapper = useCallback(
+    (width: number) => {
+      setSidebarWidth(width);
+      store.set(`${SETTING_KEY_ROOT}.fieldSelector.width`, width);
+    },
+    [setSidebarWidth]
+  );
+
+  const collapse = useCallback(() => {
+    setSidebarWidthWrapper(MIN_WIDTH);
+  }, [setSidebarWidthWrapper]);
+
+  const expand = useCallback(() => {
+    const width = getSidebarWidth(SETTING_KEY_ROOT);
+    setSidebarWidthWrapper(width < 2 * MIN_WIDTH ? DEFAULT_WIDTH : width);
+  }, [setSidebarWidthWrapper]);
+
+  const displayedColumns = useMemo(
+    () =>
+      Object.keys(columnsWithMeta)
+        .filter((column) => columnsWithMeta[column].active)
+        .sort((a, b) =>
+          columnsWithMeta[a].index !== undefined && columnsWithMeta[b].index !== undefined
+            ? columnsWithMeta[a].index - columnsWithMeta[b].index
+            : 0
+        ),
+    [columnsWithMeta]
+  );
+
+  const defaultColumns = useMemo(
+    () =>
+      Object.keys(columnsWithMeta)
+        .sort((a, b) =>
+          columnsWithMeta[a].index !== undefined && columnsWithMeta[b].index !== undefined
+            ? columnsWithMeta[a].index - columnsWithMeta[b].index
+            : 0
+        )
+        .filter(
+          (column) => columnsWithMeta[column].type === 'TIME_FIELD' || columnsWithMeta[column].type === 'BODY_FIELD'
+        ),
+    [columnsWithMeta]
+  );
+
+  const suggestedFields = useMemo(
+    () => getSuggestedFields(logs, displayedColumns, defaultColumns),
+    [defaultColumns, displayedColumns, logs]
+  );
+  const fields = useMemo(() => getFieldsWithStats(dataFrames), [dataFrames]);
+
+  return sidebarWidth > MIN_WIDTH * 2 ? (
+    <FieldSelector
+      activeFields={displayedColumns}
+      clear={clear}
+      collapse={collapse}
+      fields={fields}
+      reorder={reorder}
+      suggestedFields={suggestedFields}
+      toggle={toggle}
+    />
+  ) : (
+    <div className={logsFieldSelectorWrapperStyles.collapsedButtonContainer}>
+      <IconButton
+        className={logsFieldSelectorWrapperStyles.collapsedButton}
+        onClick={expand}
+        name="arrow-from-right"
+        tooltip={t('logs.field-selector.expand', 'Expand sidebar')}
+        size="sm"
+      />
+    </div>
+  );
+};
```

Comment:
This is cool and allows for the implementations to be different! Hopefully one day we can unify.